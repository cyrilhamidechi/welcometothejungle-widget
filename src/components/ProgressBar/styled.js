import styled from "styled-components";

import { colors } from "./../../config";

export const Wrapper = styled.div`
  width: 100%;
  height: 3px;
  background: ${colors.greyLight};
`;

export const ProgressBar = styled.div`
  width: ${({ progress }) => `${progress}%`}
  height: 100%;
  background: ${colors.green}
  transition: width 300ms ease-in-out;
`;
