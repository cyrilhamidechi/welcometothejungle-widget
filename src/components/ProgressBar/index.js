import PropTypes from 'prop-types'
import React from 'react'

import * as S from './styled'

const ProgressBar = ({ progress }) => (
  <S.Wrapper>
    <S.ProgressBar progress={progress} />
  </S.Wrapper>
)

ProgressBar.propTypes = {
  progress: PropTypes.number,
}

export default ProgressBar