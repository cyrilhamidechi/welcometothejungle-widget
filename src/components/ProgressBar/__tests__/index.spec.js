import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import ProgressBar from "../";

describe("ProgressBar", () => {
  it("should render the component", () => {
    const tree = renderer.create(<ProgressBar progress={20} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
