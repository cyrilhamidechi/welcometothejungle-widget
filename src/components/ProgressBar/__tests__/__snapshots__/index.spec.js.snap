// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should render the component 1`] = `
.c0 {
  width: 100%;
  height: 3px;
  background: #66686e;
}

.c1 {
  width: 20%;
  height: 100%;
  background: #00c29a;
  -webkit-transition: width 300ms ease-in-out;
  transition: width 300ms ease-in-out;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;
